### PR TITLE
Icon & CustomIcon component

### PR DESCRIPTION
**Pull request by jaredcwhite**
Issue bloom-housing/bloom#3248 

Some notes on the PR:

There was discussion around the icon naming since we decided to split Icon into two components, one for FontAwesome and one for bespoke SVGs. I'd proposed using Icon for FontAwesome and CustomIcon for any SVG and implemented that in this PR, but feedback is welcome.

I also tried out using Storybook Controls and separate `.tsx` files for stories in this PR. To access controls switch to the "Canvas" tab when viewing in Storybook. There's just one so far, the size prop.

Also a note regarding code style…so I've been following a lot of good thinking out there around when to (or not to) use classes in CSS and the `class` attribute vs. other sorts of attributes like data-* and aria-* and so forth for variations/states on components. Some recommended reading:

[The waster potential of attribute selectors in CSS](https://elisehe.in/2022/10/16/attribute-selectors.html)
[Style with Stateful, Semantic Selectors](https://benmyers.dev/blog/semantic-selectors/)
[Represent State with HTML Attributes, Not Class Names](https://www.aleksandrhovhannisyan.com/blog/represent-state-with-html-attributes-not-class-names/)
[CSS Classes considered harmful](https://www.keithcirkel.co.uk/css-classes-considered-harmful/)

In light of this, I thought I'd experiment here on this first component that has size variations. So instead of a class name like `icon-size-utility-sm` or `is-size-utility-sm` or whatever, I went with `[data-size="utility-sm"]`. So in the final markup, instead of `<span class="icon icon-size-utility-sm">`, it's `<span class="icon" data-size="utility-sm">`. If we wanted to _really_ go nuts, we could even wade into the waters of custom element names and attributes, for something like `<bloom-icon size="utility-sm">`, but given we're not building web components but React components, that's probably a bridge too far. I like the idea of using data-* (or when applicable, aria-*) for variations, but I welcome your feedback and discussion.

### CODE REVIEW

**Comment by jaredcwhite** [comment]
@slowbot Sure, I can update the examples to use icons from your set. And I'm also starting to waffle now if we do need them to be separate components. Might want a team confab to take a second look.

Regarding the class question, I would say this is an "internal" implantation detail so whether the component's using class or using data-*, it doesn't matter much from a component consumer standpoint because they'd use the same React prop either way.

File: src/icons/__stories__/Icon.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+<Canvas>
+  <Story id="icons-icon--preset-sizes" />
+</Canvas>
+
+## Theme Variables
+
+| Name                          | Description                | Default     |
+| ----------------------------- | -------------------------- | ----------- |
+| `--icon-utility-sm`           | sm Size                       | `0.75rem`   |
```
Comment (ludtkemorgan) [inline src/icons/__stories__/Icon.docs.mdx:62]: I might be missing some context, but how did you decide on `utility-sm`, `utility-md`, `badge-lg`, and `badge-xl`? those names don't seem intuitive to me. Also there is a big jump in size between md and lg (1rem to 2.5rem), curious how those were decided as well

**Review by ludtkemorgan — COMMENTED** [review]
I personally think that these should be one component. However, if we do want to keep them separate we could have an internal component that has all of the common logic (which is 90% of the two components) and then `CustomIcon` and `Icon` would pull that component in

**Comment by jaredcwhite** [comment]
@ludtkemorgan @slowbot All right, I've merged the two components into a single component. Essentially it seemed the easiest solution was to use an `icon` prop for the FontAwesome definitions, and just regular children for SVG nodes. Let me know what you think.

File: src/icons/Icon.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    --icon-height: var(--icon-badge-xl);
+    --icon-alignment: var(--icon-badge-xl-alignment);
+  }
+}
+
+.icon svg {
+  display: block;
+  height: 100%;
+  aspect-ratio: 1 / 1;
+  fill: currentColor;
```
Comment (emilyjablonski) [inline src/icons/Icon.scss:42]: Where does the value `currentColor` come from?

File: src/icons/Icon.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const Icon = (props: IconProps) => {
+  const classNames = ["icon"]
+  if (props.className) classNames.push(props.className)
+  const isHidden = !(props["aria-hidden"] === false)
+
+  return (
+    <span
+      id={props.id}
+      className={classNames.join(" ")}
+      data-size={props.size}
```
Comment (emilyjablonski) [inline src/icons/Icon.tsx:33]: Big fan of this over class names

**Review by emilyjablonski — APPROVED** [review]
LGTM! I know we have the size updates coming out of the huddle today but otherwise 🥳

File: src/icons/Icon.scss (hunk 2 above)
Comment (jaredcwhite) [reply src/icons/Icon.scss:42]: It's a special keyword (is that the right term ?) in CSS which means just pull in whatever the text color is. That's how we can automatically get a blue icon with blue text, red icon with red text, etc.

**Review by ludtkemorgan — APPROVED** [review]
🍨

**Pull request merged**